src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -66,6 +68,16 @@
     @Query("select pe from ProgrammingExercise pe left join fetch pe.templateParticipation tp left join fetch pe.solutionParticipation sp where tp.id = :#{#participationId} or sp.id = :#{#participationId}")
     Optional<ProgrammingExercise> findOneByTemplateParticipationIdOrSolutionParticipationId(@Param("participationId") Long participationId);
 
+    Page<ProgrammingExercise> findByTitleIgnoreCaseContainingOrCourse_TitleIgnoreCaseContaining(String partialTitle, String partialCourseTitle, Pageable pageable);
+
+    @Query("select p from ProgrammingExercise p left join fetch p.studentParticipations left join fetch p.attachments left join fetch p.categories "
+            + "left join fetch p.templateParticipation left join fetch p.solutionParticipation left join fetch p.exampleSubmissions left join fetch p.exerciseHints "
+            + "left join fetch p.tutorParticipations left join fetch p.studentQuestions left join fetch p.testCases")
+    ProgrammingExercise findOneWithEagerEverything(ProgrammingExercise exercise);
+
+    @Query("select p from ProgrammingExercise p left join fetch p.testCases left join fetch p.exerciseHints left join fetch p.templateParticipation left join fetch p.solutionParticipation where p.id = :#{#id}")
+    Optional<ProgrammingExercise> findByIdWithEagerTestCasesHintsAndBaseParticipations(Long id);
```

Comment:
same comment as above, also please do not use the generic term `id` --> always use e.g. `programmingExerciseId`

Reply:
Renamed to exerciseId